packages/clay-drop-down/src/index.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import {Align} from 'metal-position';
+import {createPortal} from 'react-dom';
+import {useDropdownCloseInteractions} from './hooks';
+
+import Action from './Action';
+import Caption from './Caption';
+import Divider from './Divider';
+import Group from './Group';
+import Help from './Help';
+import Item from './Item';
+import ItemList from './ItemList';
+import Search from './Search';
+
+const Portal = ({children}: {children: React.ReactNode}) => {
+	const portalRef = React.useRef(document.createElement('div'));
+	const elToMountTo = document && document.body;
+
+	React.useEffect(() => {
+		elToMountTo.appendChild(portalRef.current);
+		return () => {
+			elToMountTo.removeChild(portalRef.current);
+		};
+	}, [elToMountTo]);
+
+	return createPortal(children, portalRef.current);
+};
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	active: boolean;
+	alignmentPosition: number;
+	hasRightSymbols?: boolean;
+	hasLeftSymbols?: boolean;
+	onActiveChange: (val: boolean) => void;
+	trigger: React.ReactElement;
+}
+
+const ClayDropDown: React.FunctionComponent<Props> & {
+	Action: typeof Action;
+	Caption: typeof Caption;
+	Divider: typeof Divider;
+	Group: typeof Group;
+	Help: typeof Help;
+	Item: typeof Item;
+	ItemList: typeof ItemList;
+	Search: typeof Search;
+} = ({
+	active = false,
+	alignmentPosition = Align.BottomLeft,
+	children,
+	className,
+	hasLeftSymbols,
+	hasRightSymbols,
+	onActiveChange,
+	trigger,
+	...otherProps
+}) => {
+	const triggerElementRef = React.useRef<HTMLButtonElement>(null);
+	const menuElementRef = React.useRef<HTMLDivElement>(null);
+
+	useDropdownCloseInteractions(
+		[triggerElementRef, menuElementRef],
+		onActiveChange
+	);
+
+	React.useLayoutEffect(() => {
+		if (triggerElementRef.current && menuElementRef.current) {
+			Align.align(
+				menuElementRef.current,
+				triggerElementRef.current,
+				alignmentPosition
+			);
+		}
+	});
+
+	return (
+		<div {...otherProps} className="dropdown">
+			<span ref={triggerElementRef} style={{display: 'inline-block'}}>
+				{React.cloneElement(trigger, {
+					className: 'dropdown-toggle',
+					onClick: () => onActiveChange(!active),
+				})}
+			</span>
+
+			{active && (
+				<Portal>
+					<div
```

Comment:
hey @bryceosterhaus I've been thinking about making this wrapper a component too `<ClayDropDown.Menu />`, we'll have some components that we do not want to trigger for example, just your content. What do you think?

Reply:
Yeah thats a good point. I'll either look into adding that to this PR or wait and do it in another refactoring ticket that pulls it out of other components as well.

Reply:
It looks better, we can go ahead and merge this component.

Reply:
👍 I just need to update this PR with the transitions from diego's PR. Ill update today and let you know